pkg/dash/dash.go

```diff
@@ -215,15 +216,23 @@ func WithClusterClient(client cluster.ClientInterface) RunnerOption {
 	}
 }
 
+func WithStreamingClientFactory(factory api.StreamingClientFactory) RunnerOption {
```

Comment:
the package `api` in this context is an `internal` package -- for library consumers to use this option the `StreamingClientFactory` type will need to be moved into a non-internal package please!